src/main/java/ee/tuleva/onboarding/comparisons/fundvalue/retrieval/NAVCheckValueRetriever.java

```diff
@@ -0,0 +1,115 @@
+package ee.tuleva.onboarding.comparisons.fundvalue.retrieval;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import ee.tuleva.onboarding.comparisons.fundvalue.FundValue;
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.IntStream;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClient;
+import org.springframework.web.util.UriComponentsBuilder;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class NAVCheckValueRetriever implements ComparisonIndexRetriever {
+  public static final String KEY = "NAV_CHECK_VALUE";
+
+  public static final List<String> FUND_NAMES =
+      Arrays.asList("0P000152G5", "0P0001N0Z0", "SGAS.DE", "SLMC.DE", "SGAJ.DE", "0P0001MGOG");
+
+  private final RestClient.Builder restClientBuilder;
+
+  @Override
+  public String getKey() {
+    return KEY;
+  }
+
+  @Override
+  public List<FundValue> retrieveValuesForRange(LocalDate startDate, LocalDate endDate) {
+    return FUND_NAMES.stream()
+        .map(fundName -> retrieveValuesForFund(fundName, startDate, endDate))
+        .flatMap(List::stream)
+        .toList();
+  }
+
+  private List<FundValue> retrieveValuesForFund(
+      String fundName, LocalDate startDate, LocalDate endDate) {
+
+    RestClient restClient = restClientBuilder.build();
+
+    String fetchUri = buildFetchUri(fundName, startDate, endDate);
+
+    // TODO in parallel for all funds?
+    JsonNode response =
+        restClient.get().uri(fetchUri).accept(APPLICATION_JSON).retrieve().body(JsonNode.class);
+
+    JsonNode resultNode = response.path("chart").path("result").path(0);
+
+    List<LocalDate> timestamps = parseTimestamps(resultNode);
+    List<BigDecimal> fundValues = parseFundValues(resultNode);
+
+    if (fundValues.size() != timestamps.size()) {
+      throw new RuntimeException(
```

Comment:
@ErkoRisthein Earlier iteration i had a error logger here, but changed it to an exception as it would affect data quality if it still tried to add them, some fund getting invalid value etc.